client/blocks/product-selector/utils.js

```diff
@@ -10,7 +10,7 @@ import { flattenDeep, pickBy } from 'lodash';
  * @returns {Array} Array of the product slugs.
  */
 export const extractProductSlugs = ( products ) => [
-	...new Set( flattenDeep( products.map( ( product ) => Object.values( product.options ) ) ) ),
+	...new Set( products.map( ( product ) => Object.values( product.options ) ).flat( 2 ) ),
```

Comment:
Is there a difference between using `flat` or using `flatMap` in the first place?

Reply:
`flatMap` does [only one level](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap#map_and_flatmap) so it won't do the job - we're flatting 2 levels of arrays here.

Reply:
Ohhhh, are the values of `product.options` themselves arrays?

Reply:
Almost - `product.options` is an object, but `Object.values()` makes it an array.

Reply:
This `propTypes` definition makes the structure easier to understand:  https://github.com/Automattic/wp-calypso/blob/remove/lodash-union-with/client/blocks/product-selector/index.jsx#L48